gateway/httpproxy/httpproxy.go

```diff
@@ -0,0 +1,117 @@
+package httpproxy
+
+import (
+	"bufio"
+	"io"
+	"net"
+	"net/http"
+	"net/textproto"
+	"strconv"
+	"strings"
+
+	"github.com/nknorg/nkn/util/config"
+	"github.com/nknorg/nkn/util/log"
+	"github.com/xtaci/smux"
+)
+
+type HTTPProxy struct {
+	// defines a listeners for http proxy, such as "127.0.0.1:30004"
+	listener string
+}
+
+func NewServer() *HTTPProxy {
+	return &HTTPProxy{
+		listener: ":" + strconv.Itoa(int(config.Parameters.HttpProxyPort)),
+	}
+}
+
+func transfer(destination io.WriteCloser, source io.ReadCloser) {
+	defer destination.Close()
+	defer source.Close()
+	io.Copy(destination, source)
+}
+
+// parseRequestLine parses "GET /foo HTTP/1.1" into its three parts.
+func parseRequestLine(line string) (method, requestURI, proto string, ok bool) {
```

Comment:
2 questions here:
1. Shall we just use https://golang.org/pkg/strings/#SplitN here?
2. Should we consider the case were space is not escaped in url and thus the input is split into >3 parts?

Reply:
actually this part is copied from net/http, so probably it's save to leave it as it is.

Reply:
👌